GPflowOpt/acquisition/acquisition.py

```diff
@@ -273,13 +306,23 @@ def set_data(self, X, Y):
         offset = 0
         for operand in self.operands:
             offset += operand.set_data(X, Y[:, offset:])
-        if self.highest_parent == self:
-            self.setup()
         return offset
 
-    def setup(self):
+    def _setup_constraints(self):
+        for oper in self.operands:
+            if oper.constraint_indices().size > 0:
+                oper._setup_constraints() if isinstance(oper, AcquisitionAggregation) else oper.setup()
+
+    def _setup_objectives(self):
         for oper in self.operands:
-            oper.setup()
+            if oper.constraint_indices().size == 0:
+                oper._setup_objectives() if isinstance(oper, AcquisitionAggregation) else oper.setup()
```

Comment:
I think that would make the interface for implementing acquisition functions loaded with complexity which is only relevant to the internal functioning.